library/ssl_tls.c

```diff
@@ -3100,6 +3100,36 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
     }
 #endif
 
+#if defined(MBEDTLS_ECP_C)
+#if !defined(MBEDTLS_DEPRECATED_REMOVED)
+    /* Heap allocate and translate curve_list from internal to IANA group ids */
+    #define HEAP_MAX 66553
```

Comment:
This seems too large. Also, unless we use it generically as a heap allocation bound, I think we should rename this to something like `MAX_ECP_CURVE_LIST` or similar.

Reply:
Amended